### ISSUE TITLE
Understanding CRAN Incoming

### ISSUE THREAD

**Issue by elinw** [ropensci/unconf18#38]
Recently in the Slack channel I semi jokingly suggested creating an app to run a cron job to scrape the incoming folder at CRAN in order to better understand empirically how long packages stay in each state and where they move to and potentially what the bottlenecks are and what the most common kinds of problems are (for example what kinds of Notes are most common and what operating system environments have the most failures and why).  Maybe the CRAN managers already know this, but maintainers don't seem to.  Ideally if we had the data we could use it to make suggestions about how to deal with bottle necks or to improve documentation to help people avoid problems or (my pet issue) improve the understandability of notes.

**Comment by maelle** [comment]
I non-jokingly think it's a cool idea. It'd also be cool to have a dashboard because currently to find where your package is you need to click in the subfolders right?

**Comment by elinw** [comment]
Yes, that's right, it's kind of annoying that you have to keep trying different folders.

**Comment by batpigandme** [comment]
Possibly related(ish) package: François Michonneau's foghorn:
https://github.com/fmichonneau/foghorn .
The aim is different (it reports CRAN Check results), but I think it would be interesting information to integrate…

**Comment by hadley** [comment]
This is such a good idea that we have already been doing it 😉 

@edgararuiz did all the hard work, and he's going to look into sharing our scripts and the data we've collected sofar

**Comment by elinw** [comment]
@batpigandme Yes! I saw foghorn and it would be great to integrate some parts,
@hadley @edgararuiz that would be amazing!  I should have suspected you'd have thought about this.

**Comment by boshek** [comment]
Great idea. 

I'll ask, what is probably, a naive question. Is this something that we can work on *with* the folks at CRAN? This amounts to some sort of oversight of CRAN particularly if this took the form of a package. I fully understand the need for this. But I wonder if some communication with the folks at CRAN about this may useful in the long run.

**Comment by elinw** [comment]
When we have something to show them I definitely would do so.  If we find some solve-able bottlenecks I think that could help everyone.

**Comment by elinw** [comment]
That's fantastic! Lots to think about.

**Comment by elinw** [comment]
Shouldn't it also be possible to scrape the errors and notes packages generate? One of my things is that sometimes making the notes more understandable would make the whole thing less painful.  Or just to have a list of common problems would be helpful.

**Comment by elinw** [comment]
Summary: Use data science to better understand the CRAN workflow and to provide actionable information for both package maintainers and, potentially, the CRAN team about where there are bottlenecks, most common problems and other information that could make life easier for everyone.
